vllm/attention/layer.py

```diff
@@ -228,6 +233,28 @@ def forward(
                                                  value,
                                                  scale=self.scale)
             out = out.transpose(1, 2)
+        elif self.attn_backend == _Backend.HPU_ATTN:
+            from habana_frameworks.torch.hpex.kernels import FusedSDPA
+            from vllm_hpu_extension.utils import ModuleFusedSDPA
+
+            fsdpa_op = ModuleFusedSDPA(FusedSDPA)
+
+            query, key, value = (x.transpose(1, 2)
+                                 for x in (query, key, value))
+
+            out = fsdpa_op(query,
+                           key,
+                           value,
+                           None,
+                           dropout_p=0.0,
+                           is_causal=True,
```

Comment:
is encoder attention causal? I don't think it is. 
For example in _Backend.TORCH_SDPA we're calling F.scaled_dot_product_attention without specifying `is_causal` which is `False` by default